lnst/RecipeCommon/Perf/Measurements/TcRunMeasurement.py

```diff
@@ -193,23 +189,30 @@ def _make_instances_cfgs(self) -> list[TcRunInstance]:
             configs.append(cfg)
         return configs
 
-    def collect_results(self) -> list[TcRunMeasurementResults]:
-        results: list[TcRunMeasurementResults] = []
-        for job in self._finished_jobs:
-            run_result = TcRunMeasurementResults(
-                measurement=self,
-                device=self.device,
-            )
-            run_result.rule_install_rate = PerfInterval(
-                value=self._rules_per_instance,
-                duration=job.result['data']['time_taken'],
-                unit='rules',
-                timestamp=job.result['data']['start_timestamp'],
-            )
-            run_result.run_success = job.passed
-            results.append(run_result)
+    def collect_results(self):
+        run_result = TcRunMeasurementResults(
+            measurement=self,
+            device=self.device,
+        )
+        if len(self._finished_jobs) > 1:
+            raise MeasurementError("Too many jobs")
+        job = self._finished_jobs[0]
+        instance_data = job.result["data"]["instance_results"]
+
+        run_result.rule_install_rate = ParallelPerfResult(
+            [self._get_instance_interval(d) for d in instance_data]
+        )
+        run_result.run_success = all((d["success"] for d in instance_data))
```

Comment:
this can just be the `job.passed` i believe, you already calculate the `all()` value in the test module and that should transfer back into this attribute